lib/passes/PluginInfo.cpp

```diff
@@ -0,0 +1,42 @@
+#include <kllvm/codegen/MustTailDeadArgElimination.h>
+#include <kllvm/codegen/RemoveDeadKFunctions.h>
+#include <kllvm/codegen/SetVisibilityHidden.h>
+
+using namespace kllvm;
+
+/* New PM Registration */
+llvm::PassPluginLibraryInfo get_kllvm_plugin_info() {
+  return {
+      LLVM_PLUGIN_API_VERSION, "SetVisibilityHidden", LLVM_VERSION_STRING,
+      [](PassBuilder &pb) {
+        pb.registerPipelineParsingCallback(
+            [](StringRef name, llvm::ModulePassManager &pm,
+               ArrayRef<llvm::PassBuilder::PipelineElement>) {
+              if (name == "set-visibility-hidden") {
+                pm.addPass(set_visibility_hidden());
+                return true;
+              }
+              if (name == "deadargelim-musttail") {
+                pm.addPass(DeadArgumentEliminationPass());
+                return true;
+              }
+              return false;
+            });
+        pb.registerPipelineParsingCallback(
+            [](StringRef name, llvm::FunctionPassManager &pm,
+               ArrayRef<llvm::PassBuilder::PipelineElement>) {
+              if (name == "remove-dead-k-functions") {
+                pm.addPass(remove_dead_k_functions());
+                return true;
+              }
+              return false;
+            });
```

Comment:
Could you explain the way these passes are being registered here? Also why is this different than what we do in `codegen/ApplyPasses.cpp`?

Reply:
This is how the passes get registered so they run when you ask for them using the --passes flag of opt. In ApplyPasses.cpp, they are being registered as part of the default pipeline. The latter is what happens during default compilation, the former is for debugging.